pkg/workloads/console/acceptance/acceptance.go

```diff
@@ -186,6 +202,134 @@ func (r *Runner) Run(logger kitlog.Logger, config *rest.Config) {
+				return err
+			}).ShouldNot(HaveOccurred(), "could not find authorisation")
+
+			By("Expect the console phase is pending authorisation")
+			Eventually(func() workloadsv1alpha1.ConsolePhase {
+				console, err = client.WorkloadsV1Alpha1().Consoles(namespace).Get(console.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "could not find console")
+				return console.Status.Phase
+			}).Should(Equal(workloadsv1alpha1.ConsolePendingAuthorisation))
+
+			By("Expect that the job has not been created")
+			Eventually(func() error {
+				_, err = client.BatchV1().Jobs(namespace).Get(console.Name, metav1.GetOptions{})
+				return err
+			}).Should(HaveOccurred(), "expected not to find job, but did")
+
+			By("Expect a console authorisation has been created")
+			consoleAuthorisation := &workloadsv1alpha1.ConsoleAuthorisation{}
+			Eventually(func() error {
+				consoleAuthorisation, err = client.WorkloadsV1Alpha1().ConsoleAuthorisations(namespace).Get(console.Name, metav1.GetOptions{})
+				return err
+			}).ShouldNot(HaveOccurred(), "could not find console authorisation")
+
+			// Change the console user to another user as a user cannot authorise their own console
+			By("Update the console user")
+			console.Spec.User = "another-user@example.com"
+			console, err = client.WorkloadsV1Alpha1().Consoles(namespace).Update(console)
+			Expect(err).NotTo(HaveOccurred(), "could not update console user")
+
+			By("Authorise a console")
+			consoleAuthorisation.Spec.Authorisations = []rbacv1.Subject{{Kind: "User", Name: user}}
+			_, err = client.WorkloadsV1Alpha1().ConsoleAuthorisations(namespace).Update(consoleAuthorisation)
+			Expect(err).NotTo(HaveOccurred(), "could not authorise console")
+
+			By("Expect a pod has been created")
+			Eventually(func() ([]corev1.Pod, error) {
+				opts := metav1.ListOptions{LabelSelector: "console-name=" + console.Name}
+				podList, err := client.CoreV1().Pods(namespace).List(opts)
+				return podList.Items, err
+			}).Should(HaveLen(1), "expected to find a single pod")
+
+			By("Expect the console phase is Running")
+			Eventually(func() workloadsv1alpha1.ConsolePhase {
+				console, err = client.WorkloadsV1Alpha1().Consoles(namespace).Get(console.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "could not find console")
+				return console.Status.Phase
+			}).Should(Equal(workloadsv1alpha1.ConsoleRunning))
+
+			By("Expect the console phase eventually changes to Stopped")
+			Eventually(func() workloadsv1alpha1.ConsolePhase {
+				console, err = client.WorkloadsV1Alpha1().Consoles(namespace).Get(console.Name, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "could not find console")
+				return console.Status.Phase
+			}).Should(Equal(workloadsv1alpha1.ConsoleStopped))
+
+			// TODO: attach to pod
+
+			By("Expect that the console is deleted shortly after stopping, due to its TTL")
```

Comment:
I think this is something we should look into in a follow up PR. Given how small the interface of these commands has ended up being I don't see much of an issue removing the original test in favour of this one. It might also help to simplify the test a little.

Reply:
Sounds good, I think we'll naturally revisit this anyway